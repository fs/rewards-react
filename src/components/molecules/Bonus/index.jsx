import React from 'react';
import styled from 'styled-components';

const BonusContainer = styled.div`
  margin-bottom: 1.25rem;
  border-radius: 4px;
  background-color: #fff;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.1);
  color: #000;
  `;

const BonusHeader = styled.div`
  display: flex;
  align-items: flex-start;
  padding: 1.25rem 1.25rem 0;
  `;

const TotalPoints = styled.div`
  padding: 0 1rem;
  height: 2.5rem;
  line-height: 2.5rem;
  display: inline-block;
  font-size: 1.25rem;
  font-weight: bold;
  color: #63bc36;
  border-radius: 100px;
  background-color: #fff;
  border: solid 2px rgba(99, 188, 54, 0.3);
  flex-shrink: 0;

  &::before {
    display: inline-block;
    content: "+";
  }
  `;

const ReceiversList = styled.div`
  padding: 0 0.3125rem;
  width: 100%;
  `;

const ReceiverItem = styled.div`
  display: inline-block;
  width: 2.5rem;
  height: 2.5rem;
  margin: 0 0.25rem;
  border-radius: 100px;
  overflow: hidden;
  background-color: #f7f7f7;
  
  img {
    width: 100%;
  }
  `;

const Timestamp = styled.div`
  font-size: 1rem;
  color: #cecece;
  font-weight: bold;
  text-align: right;
  width: auto;
  flex-shrink: 0;
  
  span {
    color: #cecece
  }
  `;

const BonusBody = styled.div`
  padding: 0 1.25rem 1.25rem;
  `;

const TextItem = styled.div`
  margin: 10px 0 0;
  font-size: 1.25rem;
  color: #000;
  `;

const Sender = styled.span`
  font-weight: 600;
  `;

const BonusPoints = styled.span`
  font-size: 1.25rem;
  font-weight: bold;
  color: #63bc36;
  `;

const ReceiverName = styled.span`
  color: #63bc36;
  font-weight: bold;
  `;

const Tag = styled.span`
  color: #aaaaaa;
  `;

const Bonus = (props) => {
  const { bonus } = props;

  return (
    <BonusContainer data-testid="test-bonus">
      <BonusHeader>
        <TotalPoints>
          {bonus['total-points']}
        </TotalPoints>
        <ReceiversList>
          <ReceiverItem>
            <img
              src={bonus.sender.avatar}
              alt={bonus.sender.name}
            />
          </ReceiverItem>
        </ReceiversList>
        <Timestamp>
          <span title="">18 minutes ago</span>
        </Timestamp>
      </BonusHeader>
      <BonusBody>
        <TextItem>
          <Sender>
            {bonus.sender.name}
            :
            {' '}
          </Sender>
          <span>
            {
            bonus.text.map((item) => {
              if (item.type === 'count') {
                return (
                  <BonusPoints>
                    {item.text}
                    {' '}
                  </BonusPoints>
                );
              } if (item.type === 'receiver') {
                return (
                  <ReceiverName>
                    {item.text}
                    {' '}
                  </ReceiverName>
                );
              } if (item.type === 'tag') {
                return (
                  <Tag>
                    {item.text}
                    {' '}
                  </Tag>
                );
              }
              return (
                <span>
                  {item.text}
                  {' '}
                </span>
              );
            })
          }
          </span>
          {/* <BonusPoints>+{bonus.points}</BonusPoints> <ReceiverName>@marat.galeev</ReceiverName> iOS quiz <Tag>#win-win-win</Tag> */}
        </TextItem>
      </BonusBody>
    </BonusContainer>
  );
};

export default Bonus;
