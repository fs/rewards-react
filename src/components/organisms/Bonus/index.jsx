import React from 'react';
import styled from 'styled-components';
import TimeAgo from 'timeago-react';
import commentService from '../../../services/CommentService';
import CommentListItem from '../../atoms/CommentListItem';
import SendCommentForm from '../SendCommentForm';

const BonusContainer = styled.div`
  margin-bottom: 1.25rem;
  border-radius: 4px;
  background-color: #fff;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.1);
  color: #000;
`;

const BonusHeader = styled.div`
  display: flex;
  align-items: flex-start;
  padding: 1.25rem 1.25rem 0;
`;

const TotalPoints = styled.div`
  padding: 0 1rem;
  height: 2.5rem;
  line-height: 2.5rem;
  display: inline-block;
  font-size: 1.25rem;
  font-weight: bold;
  color: #63bc36;
  border-radius: 100px;
  background-color: #fff;
  border: solid 2px rgba(99, 188, 54, 0.3);
  flex-shrink: 0;

  &::before {
    display: inline-block;
    content: '+';
  }
`;

const ReceiversList = styled.div`
  padding: 0 0.3125rem;
  width: 100%;
`;

const ReceiverItem = styled.div`
  display: inline-block;
  width: 2.5rem;
  height: 2.5rem;
  margin: 0 0.25rem;
  border-radius: 100px;
  overflow: hidden;
  background-color: #f7f7f7;

  img {
    width: 100%;
  }
`;

const Timestamp = styled.div`
  font-size: 1rem;
  color: #cecece;
  font-weight: bold;
  text-align: right;
  width: auto;
  flex-shrink: 0;

  span {
    color: #cecece;
  }
`;

const BonusBody = styled.div`
  padding: 0 1.25rem 1.25rem;
`;

const TextItem = styled.div`
  margin: 10px 0 0;
  font-size: 1.25rem;
  color: #000;
`;

const Sender = styled.span`
  font-weight: 600;
`;

const BonusPoints = styled.span`
  font-size: 1.25rem;
  font-weight: bold;
  color: #63bc36;
`;

const ReceiverName = styled.span`
  color: #63bc36;
  font-weight: bold;
`;

const Tag = styled.span`
  color: #aaaaaa;
`;

const CommentList = styled.div``;

const Bonus = props => {
  const { bonus } = props;

  const onSuccess = () => {};

  return (
    <BonusContainer data-testid="test-bonus">
      <BonusHeader>
        <TotalPoints>{bonus['total-points']}</TotalPoints>
        <ReceiversList>
          {bonus.receivers.map(receiver => (
            <ReceiverItem key={receiver.email}>
              <img src={receiver['profile-image-avatar-url']} alt={receiver['full-name']} />
            </ReceiverItem>
          ))}
        </ReceiversList>
        <Timestamp>
          <TimeAgo datetime={bonus['created-at']} locale="en_US" />
        </Timestamp>
      </BonusHeader>
      <BonusBody>
        <TextItem>
          <Sender>{bonus.sender}: </Sender>
          <span>
            {bonus.text.map((item, index) => {
              if (item.type === 'points') {
                return <BonusPoints key={index}>{item.text} </BonusPoints>;
              }
              if (item.type === 'users') {
                return <ReceiverName key={index}>{item.text} </ReceiverName>;
              }
              if (item.type === 'tags') {
                return <Tag key={index}>{item.text} </Tag>;
              }
              return <span key={index}>{item.text} </span>;
            })}
          </span>
        </TextItem>
      </BonusBody>
      <CommentList data-testid="test-comment-list">
        {bonus.comments.map(item => (
          <CommentListItem comment={item} key={item.id} />
        ))}
      </CommentList>
      <SendCommentForm onSuccess={onSuccess} commentService={commentService} bonusId={bonus.id} />
    </BonusContainer>
  );
};

export default Bonus;
