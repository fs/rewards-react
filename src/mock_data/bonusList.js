export default [
  {
    id: '296',
    'created-at': '2019-05-22T08:27:24.908Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [],
    },
    sender: 'Nadezhda Kharchuk',
    receivers: [
      {
        email: 'olga.krasnova@flatstack.com',
        'full-name': 'Olga Krasnova',
        username: 'krasnova-olga',
        'profile-image-avatar-url': 'https://d14n4flg6h47l7.cloudfront.net/store/user/62/profile_image/avatar-9c17bcabdb06f23cc9c93a1d6994bc1a.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIJWLCQQNSWJW7QQA%2F20190522%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190522T082725Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=31db982a9259d275e2493bbc8a221e8d47cf5cfa6934f94c31851c107bc09d0e',
        'bonus-balance': 10078,
        'allowance-balance': 489,
      },
    ],
    text: [
      {
        text: '+cat',
        type: 'text',
      },
      {
        text: '+1',
        type: 'points',
      },
      {
        text: '@blah',
        type: 'text',
      },
      {
        text: 'and',
        type: 'text',
      },
      {
        text: '-20',
        type: 'text',
      },
      {
        text: '@fake',
        type: 'text',
      },
      {
        text: 'hahahaha',
        type: 'text',
      },
      {
        text: '#90',
        type: 'tags',
      },
      {
        text: '+1+345',
        type: 'text',
      },
      {
        text: '#communicate-transparently-honestly',
        type: 'tags',
      },
      {
        text: '@krasnova-olga',
        type: 'users',
      },
      {
        text: '',
        type: 'text',
      },
    ],
  },
  {
    id: '295',
    'created-at': '2019-05-22T07:47:43.175Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [
        {
          id: '365',
          type: 'comments',
        },
      ],
    },
    sender: 'Nadezhda Kharchuk',
    receivers: [
      {
        email: 'olga.krasnova@flatstack.com',
        'full-name': 'Olga Krasnova',
        username: 'krasnova-olga',
        'profile-image-avatar-url': 'https://d14n4flg6h47l7.cloudfront.net/store/user/62/profile_image/avatar-9c17bcabdb06f23cc9c93a1d6994bc1a.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIJWLCQQNSWJW7QQA%2F20190522%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190522T082725Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=31db982a9259d275e2493bbc8a221e8d47cf5cfa6934f94c31851c107bc09d0e',
        'bonus-balance': 10078,
        'allowance-balance': 489,
      },
    ],
    text: [
      {
        text: '+1',
        type: 'points',
      },
      {
        text: '@krasnova-olga',
        type: 'users',
      },
      {
        text: '+1',
        type: 'text',
      },
      {
        text: '+1',
        type: 'text',
      },
      {
        text: '+1',
        type: 'text',
      },
      {
        text: '#set-people-up-to-succeed',
        type: 'tags',
      },
      {
        text: 'test',
        type: 'text',
      },
    ],
  },
  {
    id: '294',
    'created-at': '2019-05-22T07:47:08.999Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [],
    },
    sender: 'Nadezhda Kharchuk',
    receivers: [
      {
        email: 'alexey.toksarov@flatstack.com',
        'full-name': 'Alexey Toksarov',
        username: 'alexey.toksarov',
        'profile-image-avatar-url': 'http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png',
        'bonus-balance': 5218,
        'allowance-balance': 500,
      },
    ],
    text: [
      {
        text: '@tes',
        type: 'text',
      },
      {
        text: '@alexey.toksarov',
        type: 'users',
      },
      {
        text: '+1',
        type: 'points',
      },
      {
        text: '+2',
        type: 'text',
      },
      {
        text: '+3',
        type: 'text',
      },
      {
        text: '#be-curious-never-stop-learning',
        type: 'tags',
      },
      {
        text: '',
        type: 'text',
      },
    ],
  },
  {
    id: '293',
    'created-at': '2019-05-22T07:46:15.286Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [],
    },
    sender: 'Nadezhda Kharchuk',
    receivers: [
      {
        email: 'olga.krasnova@flatstack.com',
        'full-name': 'Olga Krasnova',
        username: 'krasnova-olga',
        'profile-image-avatar-url': 'https://d14n4flg6h47l7.cloudfront.net/store/user/62/profile_image/avatar-9c17bcabdb06f23cc9c93a1d6994bc1a.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIJWLCQQNSWJW7QQA%2F20190522%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190522T082725Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=31db982a9259d275e2493bbc8a221e8d47cf5cfa6934f94c31851c107bc09d0e',
        'bonus-balance': 10078,
        'allowance-balance': 489,
      },
    ],
    text: [
      {
        text: '@test',
        type: 'text',
      },
      {
        text: '+1',
        type: 'points',
      },
      {
        text: '@krasnova-olga',
        type: 'users',
      },
      {
        text: '#be-curious-never-stop-learning',
        type: 'tags',
      },
      {
        text: '',
        type: 'text',
      },
    ],
  },
  {
    id: '292',
    'created-at': '2019-05-21T08:37:37.644Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [],
    },
    sender: 'Olga Krasnova',
    receivers: [
      {
        email: 'nadezhda.kharchuk@flatstack.com',
        'full-name': 'Nadezhda Kharchuk',
        username: 'nadezhda.kharchuk',
        'profile-image-avatar-url': 'https://d14n4flg6h47l7.cloudfront.net/store/user/39/profile_image/avatar-58e3224793345ae993f75a6f373f3567.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIJWLCQQNSWJW7QQA%2F20190522%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190522T082724Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=c227c2bcbf1b978044d3d6370d0f9708d2319c9ec5de7ca812c424a2d7dcb15f',
        'bonus-balance': 79,
        'allowance-balance': 471,
      },
    ],
    text: [
      {
        text: '@nadezhda.kharchuk',
        type: 'users',
      },
      {
        text: '+1',
        type: 'points',
      },
      {
        text: 'Hey',
        type: 'text',
      },
      {
        text: '#be-curious-never-stop-learning',
        type: 'tags',
      },
      {
        text: '',
        type: 'text',
      },
    ],
  },
  {
    id: '291',
    'created-at': '2019-05-21T08:35:55.120Z',
    points: 5,
    'total-points': 5,
    comments: {
      data: [],
    },
    sender: 'Nadezhda Kharchuk',
    receivers: [
      {
        email: 'john.travolta@gmail.com',
        'full-name': 'John Travolta',
        username: 'john.travolta',
        'profile-image-avatar-url': 'http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png',
        'bonus-balance': 5,
        'allowance-balance': 500,
      },
      {
        email: 'leyla.khamidullina@flatstack.com',
        'full-name': 'Leyla Khamidullina',
        username: 'leyla.khamidullina',
        'profile-image-avatar-url': 'http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png',
        'bonus-balance': 50084,
        'allowance-balance': 500,
      },
      {
        email: 'albert.fazullin@flatstack.com',
        'full-name': 'Albert Fazullin',
        username: 'albert.fazullin',
        'profile-image-avatar-url': 'http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png',
        'bonus-balance': 62,
        'allowance-balance': 500,
      },
    ],
    text: [
      {
        text: '@john.travolta',
        type: 'users',
      },
      {
        text: '@albert.fazullin',
        type: 'users',
      },
      {
        text: '@leyla.khamidullina',
        type: 'users',
      },
      {
        text: '+5',
        type: 'points',
      },
      {
        text: '#be-curious-never-stop-learning',
        type: 'tags',
      },
      {
        text: '',
        type: 'text',
      },
    ],
  },
  {
    id: '290',
    'created-at': '2019-05-21T08:33:00.539Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [],
    },
    sender: 'Nadezhda Kharchuk',
    receivers: [
      {
        email: 'olga.krasnova@flatstack.com',
        'full-name': 'Olga Krasnova',
        username: 'krasnova-olga',
        'profile-image-avatar-url': 'https://d14n4flg6h47l7.cloudfront.net/store/user/62/profile_image/avatar-9c17bcabdb06f23cc9c93a1d6994bc1a.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIJWLCQQNSWJW7QQA%2F20190522%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20190522T082725Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=31db982a9259d275e2493bbc8a221e8d47cf5cfa6934f94c31851c107bc09d0e',
        'bonus-balance': 10078,
        'allowance-balance': 489,
      },
    ],
    text: [
      {
        text: '+1',
        type: 'points',
      },
      {
        text: '@krasnova-olga',
        type: 'users',
      },
      {
        text: '#be-curious-never-stop-learning',
        type: 'tags',
      },
      {
        text: '',
        type: 'text',
      },
    ],
  },
  {
    id: '289',
    'created-at': '2019-05-16T06:07:25.102Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [],
    },
    sender: 'bot',
    receivers: [
      {
        email: 'ivan.ananev@flatstack.com',
        'full-name': 'Ivan Ananev',
        username: 'ivan.ananev',
        'profile-image-avatar-url': 'http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png',
        'bonus-balance': 6,
        'allowance-balance': 500,
      },
    ],
    text: [
      {
        text: '+1',
        type: 'points',
      },
      {
        text: 'to',
        type: 'text',
      },
      {
        text: '@ivan.ananev',
        type: 'users',
      },
      {
        text: '#be-curious-never-stop-learning',
        type: 'tags',
      },
      {
        text: 'for',
        type: 'text',
      },
      {
        text: 'move',
        type: 'text',
      },
    ],
  },
  {
    id: '284',
    'created-at': '2019-04-30T10:45:31.860Z',
    points: 1,
    'total-points': 1,
    comments: {
      data: [
        {
          id: '355',
          type: 'comments',
        },
      ],
    },
    sender: 'Mariya Valeeva',
    receivers: [
      {
        email: 'almaz.karimullin@flatstack.com',
        'full-name': 'Almaz Karimullin',
        username: 'almaz.karimullin',
        'profile-image-avatar-url': 'http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png',
        'bonus-balance': 176,
        'allowance-balance': 500,
      },
    ],
    text: [
      {
        text: '#test',
        type: 'tags',
      },
      {
        text: '',
        type: 'text',
      },
      {
        text: '+1',
        type: 'points',
      },
      {
        text: '',
        type: 'text',
      },
      {
        text: '@almaz.karimullin',
        type: 'users',
      },
      {
        text: '',
        type: 'text',
      },
    ],
  },
  {
    id: '283',
    'created-at': '2019-04-30T08:08:48.835Z',
    points: 5,
    'total-points': 5,
    comments: {
      data: [],
    },
    sender: 'Nadezhda Kharchuk',
    receivers: [
      {
        email: 'albert.fazullin@flatstack.com',
        'full-name': 'Albert Fazullin',
        username: 'albert.fazullin',
        'profile-image-avatar-url': 'http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png',
        'bonus-balance': 62,
        'allowance-balance': 500,
      },
    ],
    text: [
      {
        text: '#be-curious-never-stop-learning',
        type: 'tags',
      },
      {
        text: '@albert.fazullin',
        type: 'users',
      },
      {
        text: '+5',
        type: 'points',
      },
      {
        text: 'test',
        type: 'text',
      },
    ],
  },
];
