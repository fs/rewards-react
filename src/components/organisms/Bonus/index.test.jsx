import 'jest-dom/extend-expect';
import ShallowRenderer from 'react-test-renderer/shallow';

import React from 'react';

import Bonus from '.';
import bonusList from '../../../mock_data/bonusList';

describe('Bonus index', () => {
  test('render Bonus correctly', () => {
    // Arrange
    const renderer = new ShallowRenderer();
    renderer.render(<Bonus bonus={bonusList[0]} />);
    // Act
    const result = renderer.getRenderOutput();

    // Assert
    expect(result).toMatchSnapshot();
  });
});
