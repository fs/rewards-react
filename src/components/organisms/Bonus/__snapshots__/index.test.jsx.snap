// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bonus index render Bonus correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gZMcBi iRXrkI"
        data-testid="test-bonus"
      >
        <div
          class="sc-gqjmRU eiKwRT"
        >
          <div
            class="sc-VigVT iAALtE"
          >
            1
          </div>
          <div
            class="sc-jTzLTM golyQb"
          >
            <div
              class="sc-fjdhpX jRhsDE"
            >
              <img
                alt="Albert Fazullin"
                src="http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png"
              />
            </div>
          </div>
          <div
            class="sc-jzJRlG EtLpg"
          />
        </div>
        <div
          class="sc-cSHVUG jUUdoM"
        >
          <div
            class="sc-kAzzGY fMvBW"
          >
            <span
              class="sc-chPdSV flxTt"
            >
              Nadezhda Kharchuk
              : 
            </span>
            <span>
              <span
                class="sc-kgoBCf wUPVw"
              >
                +1
                 
              </span>
              <span
                class="sc-kGXeez dKGuCC"
              >
                @albert.fazullin
                 
              </span>
              <span
                class="sc-kpOJdX eAfyei"
              >
                #be-curious-never-stop-learning
                 
              </span>
            </span>
          </div>
        </div>
        <div
          class="sc-dxgOiQ hTjMfW"
          data-testid="test-comment-list"
        />
        <form
          class="sc-gzVnrw pJiTn"
          data-testid="test-bonus-form"
          tabindex="0"
        >
          <div
            data-testid="test-textarea-wrapper"
          >
            <div
              class="rta  "
            >
              <textarea
                class="rta__textarea autocomplete-textarea autocomplete-textarea--comment"
                data-testid="test-textarea"
                name="bonus-text"
                placeholder="Add a comment"
              />
            </div>
          </div>
          <div
            data-testid="test-error-container"
          />
          <div
            class="sc-iwsKbI guaXQX"
          >
            <button
              class="sc-EHOje jlYtND"
              data-testid="test-button"
            >
              Add
            </button>
            <div
              class="sc-dnqmqq iFsitv"
            >
              <button
                class="sc-bZQynM hzcfhx"
                type="button"
              >
                <img
                  alt="Points"
                  src="[object Object]"
                />
              </button>
              <button
                class="sc-bZQynM hzcfhx"
                type="button"
              >
                <img
                  alt="Hashtag"
                  src="[object Object]"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gZMcBi iRXrkI"
      data-testid="test-bonus"
    >
      <div
        class="sc-gqjmRU eiKwRT"
      >
        <div
          class="sc-VigVT iAALtE"
        >
          1
        </div>
        <div
          class="sc-jTzLTM golyQb"
        >
          <div
            class="sc-fjdhpX jRhsDE"
          >
            <img
              alt="Albert Fazullin"
              src="http://doypx5fd26upp.cloudfront.net/assets/default-user-profile_image-b50dbea6a3e28a1ae62538d4b0115b18558fac3ff7dbbee7da800b5216f97070.png"
            />
          </div>
        </div>
        <div
          class="sc-jzJRlG EtLpg"
        />
      </div>
      <div
        class="sc-cSHVUG jUUdoM"
      >
        <div
          class="sc-kAzzGY fMvBW"
        >
          <span
            class="sc-chPdSV flxTt"
          >
            Nadezhda Kharchuk
            : 
          </span>
          <span>
            <span
              class="sc-kgoBCf wUPVw"
            >
              +1
               
            </span>
            <span
              class="sc-kGXeez dKGuCC"
            >
              @albert.fazullin
               
            </span>
            <span
              class="sc-kpOJdX eAfyei"
            >
              #be-curious-never-stop-learning
               
            </span>
          </span>
        </div>
      </div>
      <div
        class="sc-dxgOiQ hTjMfW"
        data-testid="test-comment-list"
      />
      <form
        class="sc-gzVnrw pJiTn"
        data-testid="test-bonus-form"
        tabindex="0"
      >
        <div
          data-testid="test-textarea-wrapper"
        >
          <div
            class="rta  "
          >
            <textarea
              class="rta__textarea autocomplete-textarea autocomplete-textarea--comment"
              data-testid="test-textarea"
              name="bonus-text"
              placeholder="Add a comment"
            />
          </div>
        </div>
        <div
          data-testid="test-error-container"
        />
        <div
          class="sc-iwsKbI guaXQX"
        >
          <button
            class="sc-EHOje jlYtND"
            data-testid="test-button"
          >
            Add
          </button>
          <div
            class="sc-dnqmqq iFsitv"
          >
            <button
              class="sc-bZQynM hzcfhx"
              type="button"
            >
              <img
                alt="Points"
                src="[object Object]"
              />
            </button>
            <button
              class="sc-bZQynM hzcfhx"
              type="button"
            >
              <img
                alt="Hashtag"
                src="[object Object]"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
